front/app/containers/ProjectsShowPage/shared/header/ProjectActionButtons.tsx

```diff
@@ -131,54 +132,42 @@ const ProjectActionButtons = memo<Props>(({ projectId, className }) => {
 
   const { process_type, publication_status } = project.data.attributes;
 
-  const isProjectArchived =
-    project.data.attributes.publication_status === 'archived';
+  const isProjectArchived = publication_status === 'archived';
   const isProcessTypeContinuous = process_type === 'continuous';
-  const participation_method = isProcessTypeContinuous
+  const isProcessTypeTimeline = process_type === 'timeline';
+  const participationMethod = isProcessTypeContinuous
     ? project.data.attributes.participation_method
     : currentPhase?.attributes.participation_method;
   const ideas_count = isProcessTypeContinuous
     ? project.data.attributes.ideas_count
     : currentPhase?.attributes.ideas_count;
-  const hasProjectEnded = currentPhase
+  const hasCurrentPhaseEnded = currentPhase
     ? pastPresentOrFuture([
         currentPhase.attributes.start_at,
         currentPhase.attributes.end_at,
       ]) === 'past'
     : false;
-  const inputTerm = getInputTerm(
-    project.data.attributes.process_type,
-    project.data,
-    phases?.data
-  );
-
-  const isParticipationMethodIdeation = participation_method === 'ideation';
-
+  const inputTerm = getInputTerm(process_type, project.data, phases?.data);
   const isParticipationMethodNativeSurvey =
-    participation_method === 'native_survey' ||
-    currentPhase?.attributes.participation_method === 'native_survey';
+    participationMethod === 'native_survey';
+  const isParticipationMethodIdeation = participationMethod === 'ideation';
 
   const showSeeIdeasButton =
-    ((isProcessTypeContinuous && isParticipationMethodIdeation) ||
-      currentPhase?.attributes.participation_method === 'ideation') &&
-    isNumber(ideas_count) &&
-    ideas_count > 0;
+    isParticipationMethodIdeation && isNumber(ideas_count) && ideas_count > 0;
 
-  const showIdeasButton =
-    isParticipationMethodIdeation && publication_status !== 'archived';
+  const showIdeasButton = isParticipationMethodIdeation && !isProjectArchived;
 
   const showNativeSurvey =
-    isParticipationMethodNativeSurvey && publication_status !== 'archived';
+    isParticipationMethodNativeSurvey && !isProjectArchived;
 
   const showSurvey =
-    (!isProjectArchived && phases && participation_method === 'survey') ||
-    (currentPhase?.attributes.participation_method === 'survey' &&
-      !hasProjectEnded);
+    participationMethod === 'survey' &&
+    (!isProjectArchived || !hasCurrentPhaseEnded);
```

Comment:
In the old code for showSurvey, we basically said: show if proj is not archived and we have a survey (either continuous proj or in a phase/timeline) OR if we have a survey phase (already checked if we check for participationMethod) that has not ended.

In the new code, we do the same thing: show if method is survey and the project is not archived or the phase has not ended. If we have a continuous project, current phase is null, and hasCurrentPhaseEnded will evaluate to false, so !false equals true and !hasCurrentPhaseEnded is always true in a continuous project. 

Ideally, we split this code up (one for timeline, one for continuous), but that would be too much work for now.